opaque_keys/edx/locator.py

```diff
@@ -1220,14 +1220,15 @@ def _from_string(cls, serialized):
 
         return cls(**{key: parse.get(key) for key in cls.KEY_FIELDS})
 
+    @property
```

Comment:
This would need to be documented as a `BREAKING CHANGE`, and I'm not clear on its implications.  Do you know who this will affect and how big of an effect it will be?  Maybe it is negligible?  I have no idea. :)

Can you provide what you think would go in the commit message around this, and do you feel confident or should we have someone else check this?

Reply:
There's an example of the effect at https://github.com/edx/opaque-keys/pull/116/files#diff-799b9a5aa15972948c85ca0d093c01d3R43

The changelog major version bump documents it, but perhaps not loudly enough. :-)

Based on my limited understanding of Python properties and methods, this seems like a bugfix, and maybe it's just a bug that no one has run into yet. When I look around the edx org, I only find a few obvious references to DefinitionLocator, but of course someone else could be using it, and they might want it to be bug-compatible. Not really sure where to go with that.

Reply:
Specifically, the superclass `Locator` declares
```
    @abstractproperty
    def version(self):
```
so anything using a DefinitionLocator via the Locator interface would be in for a surprise if it called `locator.version` rather than `locator.version()`.

Reply:
It looks like Opaque Keys job was already fixed, so this feels quite independent.  If you want to shepherd it through and to some reviews of other code to ensure this won't break edx-platform, that is up to you.  It seems like a change with some risk, so that risk should be well documented and you might want to be the one to update edx-platform so you are on call when and if it breaks things.

Reply:
@ormsbee Do you have any thoughts on if there are particular places I should check for breaking changes?

I'm also going to see if this breaks any tests in edx-platform, although I'll have to work around edx-ccx-keys having pinned `edx-opaque-keys<3.0.0`.

Reply:
[No failures on edx-platform](https://github.com/edx/edx-platform/pull/23447), although I notice something else has changed with the resolved dependencies; it was previously pulling in `edx-opaque-keys[django]` and now doesn't. I guess removing the `extras_require` means that this project is no longer declaring a `[django]` extra, and I'm not sure what the effect of that is—or how I would make it mesh with our pip-tools approach if we wanted to keep it.

Reply:
Honestly, opaque-keys is old enough and widely used enough (in a bunch of random places), that I wouldn't pick any battles that don't have to be fought. But if you're looking at where this get exercised (if at all), my guess would be that it's happening in the the Split modulestore (look for `DraftVersioningModuleStore` and its parent classes).

Please feel free to change `edx-ccx-keys` as you need as well.

Reply:
Hmm. I give I'll leave it alone for now, and pull out the requirements files part as a separate PR.